salt/pillar/consul_pillar.py

```diff
@@ -258,7 +258,13 @@ def consul_fetch(client, path):
     '''
     Query consul for all keys/values within base path
     '''
-    return client.kv.get(path, recurse=True)
+    # Unless the root path is blank, it needs a trailing slash for
+    # the kv get from Consul to work as expected
+    if not path:
+        absolute_path = path
+    else:
+        absolute_path = path.rstrip('/') + '/'
+    return client.kv.get(absolute_path, recurse=True)
```

Comment:
Almost. :wink:  If you are not sure `path` may be `None`, it might still crash. So:
```python
return client.kv.get(os.path.join(path or ''.rstrip(os.path.sep), ''), recurse=True)
```

Otherwise just `.rstrip` it.

Reply:
Ah, right for back-slash. Still can be much less coding:
```python
return client.kv.get((path or '').rstrip('/') + '/', recurse=True)
```